bindings/nodejs/lib/types/block/public-key.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+// Temp solution for not double parsing JSON
+import { HexEncodedString } from '../utils';
+
+/**
+ * All of the public key types.
+ */
+enum PublicKeyType {
+    Ed25519 = 0,
+}
+
+/**
+ * A public key.
+ */
+abstract class PublicKey {
+    readonly type: PublicKeyType;
+
+    constructor(type: PublicKeyType) {
+        this.type = type;
+    }
+}
+
+/**
+ * Ed25519 public key.
+ */
+class Ed25519PublicKey extends PublicKey {
+    /**
+     * The public key.
+     */
+    publicKey: HexEncodedString;
+
+    constructor(publicKey: HexEncodedString) {
+        super(PublicKeyType.Ed25519);
+        this.publicKey = publicKey;
+    }
+}
+
+export { Ed25519PublicKey, PublicKey, PublicKeyType };
```

Comment:
Do we not need the `Discriminator` thing? It's not very clear to me what it does and when we need it

Reply:
Also not clear to me if it's needed, especially if there is only a single variant atm 
@kwek20 do we need it?

Reply:
If I'm not mistaken,  the Discriminator let the class transformer decides what class should it be used to create the intance depending on some value, usually the `type` property for example: 

```Typescript
const FeatureDiscriminator = {
    property: 'type',
    subTypes: [
        { value: SenderFeature, name: FeatureType.Sender as any },
        { value: IssuerFeature, name: FeatureType.Issuer as any },
        { value: MetadataFeature, name: FeatureType.Metadata as any },
        { value: TagFeature, name: FeatureType.Tag as any },
        { value: BlockIssuerFeature, name: FeatureType.BlockIssuer as any },
        { value: StakingFeature, name: FeatureType.Staking as any },
    ],
};
``` 
if the property `type` == FeatureType.Sender (which is 0) it will create an instance of class `SenderFeature` and so on.

Reply:
Why is it `name:` and not `type:`?

Reply:
That I don't know 😆

Reply:
it has a field: `property: 'type',` which sets based on which json property it runs. `value: and name:` are just parts of this class-transformer, its its values